klp_tc_functions.sh

```diff
@@ -183,7 +185,24 @@ function klp_wait_complete() {
         (( TIMEOUT-- )) || true
     done
 
-    ! klp_in_progress
+    if klp_in_progress; then
+        return 1
+    fi
+
+    # See if the livepatch in question had been disabled and wait for
+    # the module's reference count to drop asynchronously to zero if
+    # so.
```

Comment:
Nicolai, I like the module refcount watch enhancement but I'm not sure if `klp_wait_complete` is the right home for this code. Two subjective, or "code clarity" if that sounds better, arguments for that.

1) `klp_wait_complete` is looking into future. In most cases, we insert a live patch and wait until the transition completes. So, `klp_wait_complete my_live_patch 30` reads to me as "wait until transition to my_live_patch completes and fail if that takes longer than 30 seconds". However, that is not what this code bit does. It checks that transition **from** `my_live_patch` completes and does nothing in case of migration **to* the live patch in question.
2) The above confusing behavior perhaps demonstrates in the `klp_tc_3.sh` change. I chose the test as it's the simplest one. There we are inserting the `getpid()` patch and we are waiting until it's in use. So, `klp_wait_complete` does not need the extra argument. We are removing the `getpid()` patch automatically in `klp_tc_exit`. That's where we need to wait for the module refcount to settle.

That said, it looks to me like, the module refcount checking should be a new function that is applied at places where we are removing kernel modules, or deactivating patches. Agree?